tdb 1.4.10 (new formula)

kinkie opened the pull request:
tdb (Trivial DataBase) is a library with accompanying utilities,
built by the Samba team to work as a simple, synchronized
key-value store, similar to gdbm, Berkeley db and others,
with additional features for safe concurrent access.

<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

kinkie commented:
The failed build test is due to errors in the libtirpc formula, which is pulled in via the
python build dependency. I'm not sure how to deal with this right now. Any suggestions?

samford commented:
The `libtirpc` attestation failure is something that we have to resolve on our end (it affects a number of other PRs as well). We've disabled attestations on CI for the moment (we enabled it only yesterday), so I've rebased this to pick up the workflow change and re-run it.

kinkie commented:
> While I was at it, I squashed the commits (and removed the trailing period from the first commit title), so it will pass the commit style check.

Thanks @samford ! The style check failure cause wasn't clear to me

chenrui333 commented on Formula/t/tdb.rb:
```diff
@@ -0,0 +1,27 @@
+  license "GPL-3.0-or-later"
+
+  uses_from_macos "python" => :build
+  on_linux do
+    depends_on "python@3.12" => :build
+  end
```
this would be redundant?

kinkie replied:
Tdb needs to be able to run python to build. The formula wasn't working until I added it as a dependency. On macOS I can rely on the system one, but it seems that at least on the GitHub runners it needs to be explicitly specified

branchvincent replied:
makes sense but `uses_from_macos` is sufficient to also install python on linux (since it is functionally equivalent to this `on_linux` block), so we can remove it

kinkie replied:
Removed, confirmed working. Thanks!

chenrui333 commented on Formula/t/tdb.rb:
```diff
@@ -0,0 +1,27 @@
+  end
+
+  def install
+    system "./configure", "--prefix=#{prefix}"
```
would this work?
```suggestion
    system "./configure", *std_configure_args
```

kinkie replied:
No. Tdb doesn't use autotools but waf as a build chain. It doesn't recognise the standard arguments

p-linnane reviewed (approved):
Thanks @kinkie!

kinkie commented:
> Thanks @kinkie!

Thank you all for the great review experience!
